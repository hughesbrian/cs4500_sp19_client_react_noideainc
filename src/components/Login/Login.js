import React from 'react'

const Login = ({email, password, updateField, login}) =>
    <div className="container">
        <h1>Welcome back</h1>
        <br/>
        <div>
            <div className="row">
                <div className="col-12">
                    <label htmlFor ="email">Email</label>
                    <input onChange={updateField} value={email} id="email" name="email" className="form-control"/>
                </div>
                <div className="col-12">
                    <br/>
                    <label htmlFor ="password">Password</label>
                    <input onChange={updateField} value={password} id="password" name="password" className="form-control"/>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-6">
                    <label htmlFor="remember">
                        <input type="checkbox"/>
                        &nbsp; Remember me
                    </label>
                </div>
                <div className="col-6">
                    <a className="float-right" href="#">Forgot password?</a>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-12">
                    <a onClick={login} className="btn btn-primary btn-block">
                        Log in
                    </a>
                </div>
            </div>
        </div>
    </div>

export default Login