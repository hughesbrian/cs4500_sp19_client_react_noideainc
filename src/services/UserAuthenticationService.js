const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:8080' : 'https://cs4500-sp19-noideainc.herokuapp.com'

export default class UserAuthenticationService {
    static instance = null;

    static getInstance() {
        if (UserAuthenticationService.instance === null) {
            UserAuthenticationService.instance = new UserAuthenticationService()
        }
        return this.instance
    }

    login = user => 
        fetch(baseURL + "/api/login", 
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'},
                credentials: "include"
            }
        ).then(response => {
            return response.json()
        })

    logout = () => 
        fetch(baseURL + "/api/logout", 
            {
                method: 'POST',
                headers: {'content-type': 'application/json'},
                credentials: "include"
            }
        ).then(response => response.json())

    checkLogin = () => 
        fetch(baseURL + "/api/checkLogin", {
            method: 'GET',
            credentials: "include"
        })
        .then(response => {
            return response.json()
        })

}