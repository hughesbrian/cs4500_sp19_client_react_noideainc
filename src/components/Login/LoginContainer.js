import React from 'react'
import Login from '../../components/Login/Login'
import userAnthentication from '../../services/UserAuthenticationService'

class LoginContainer extends React.Component {
    constructor(props) {
        super(props);
        this.userAnthentication = userAnthentication.getInstance();
        this.state = {
            email: "",
            password: ""
        }
    }

    updateField = (e) => {
        this.setState({
            [e.target.name] : e.target.value 
        })
    }

    login = () => {
        console.log("log in")
        if (this.state.email != "" && this.state.password != "") {
            this.userAnthentication
            .login({
                email: this.state.email,
                password: this.state.password
            })
            .then((response) => {
                //console.log(response)
                if (response != null && response.email != null && response.password != null) {
                    alert("Log in Successfully")
                    window.location.reload()
                    this.props.history.push("/home")
                }
            }).catch((error) => {
                console.log(error)
                alert("Please input correct email and password");
                this.setState({
                    email: "",
                    password: ""
                })
            });
        } else {
            alert("Please input email and password");
        }
    }

    render = () =>
        <Login 
            email={this.state.email}
            password={this.state.password}
            updateField={this.updateField}
            login={this.login}
        />
}

export default LoginContainer;