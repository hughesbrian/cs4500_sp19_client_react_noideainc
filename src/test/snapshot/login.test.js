import React from 'react'
import Login from '../../components/Login/Login'
import TestRenderer from 'react-test-renderer'
import users from "../MockData/Users1.mock.json"   
import '../mockservice/UserAuthenticationService.mock'

test('test login', () => {
    const login = () => {

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot()
        
        return users[0]

    }

    const testRenderer = TestRenderer.create(
        <Login 
            email={users[0].email}
            password={users[0].password}
            login={login}
        />
    )

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root

    const loginButton = testInstance.findByProps({className: 
    'col-12 login'})

    loginButton.props.onClick()

})

test('test render login page correctly', () => {

    const testRenderer = TestRenderer.create(
        <Login 
            email={users[0].email}
            password={users[0].password}
        />
    )

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root

    const email = testInstance.findByProps({className: "form-control email"})
    const password = testInstance.findByProps({className: "form-control password"})
    const rows = testInstance.findAllByProps({className: "row"})
    const labels = testInstance.findAllByProps({className: "label"})

    expect(email.props.value).toBe('nate@gamil.com')
    expect(password.props.value).toBe('p4ssw0rd')
    expect(rows.length).toBe(3)
    expect(labels.length).toBe(3)

})