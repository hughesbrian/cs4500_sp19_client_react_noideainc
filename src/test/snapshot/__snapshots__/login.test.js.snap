// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login 1`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control email"
          id="email"
          name="email"
          value="nate@gamil.com"
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          className="label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control password"
          id="password"
          name="password"
          value="p4ssw0rd"
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          className="label"
          htmlFor="remember"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12 login"
        onClick={[Function]}
      >
        <a
          className="btn btn-primary btn-block"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`test render login page correctly 1`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          className="label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control email"
          id="email"
          name="email"
          value="nate@gamil.com"
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          className="label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control password"
          id="password"
          name="password"
          value="p4ssw0rd"
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          className="label"
          htmlFor="remember"
        >
          <input
            type="checkbox"
          />
            Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12 login"
      >
        <a
          className="btn btn-primary btn-block"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
</div>
`;
